java/delete-features-feature-service/src/main/java/com/esri/arcgisruntime/samples/deletefeaturesfeatureservice/MainActivity.java

```diff
@@ -0,0 +1,238 @@
+    QueryParameters queryParameters = new QueryParameters();
+    queryParameters.setWhereClause(String.format("OBJECTID = %s", featureId));
+
+    ListenableFuture<FeatureQueryResult> featureQueryResult = mFeatureLayer.getFeatureTable()
+        .queryFeaturesAsync(queryParameters);
+    featureQueryResult.addDoneListener(() -> {
+      try {
+        // check result has a feature
+        if (featureQueryResult.get().iterator().hasNext()) {
+          // attempt to get first feature from result as it should be the only feature
+          Feature foundFeature = featureQueryResult.get().iterator().next();
+          // delete found features
+          deleteFeature(foundFeature, mFeatureTable, () -> {
+            applyEdits(mFeatureTable);
+          });
+        }
+      } catch (InterruptedException | ExecutionException e) {
+        logToUser(getString(R.string.error_feature_deletion, e.getCause().getMessage()));
+      }
+    });
+  }
+
+  /**
+   * Deletes a feature from a {@link ServiceFeatureTable} and applies the changes to the
+   * server.
+   *
+   * @param feature                     {@link Feature} to delete
+   * @param featureTable                {@link ServiceFeatureTable} to delete {@link Feature} from
+   * @param onDeleteFeatureDoneListener {@link Runnable} to be invoked when action has completed
+   */
+  private void deleteFeature(Feature feature, ServiceFeatureTable featureTable,
+      Runnable onDeleteFeatureDoneListener) {
+    // delete feature from the feature table and apply edit to server
+    featureTable.deleteFeatureAsync(feature).addDoneListener(onDeleteFeatureDoneListener);
+  }
+
+  /**
+   * Sends any edits on the {@link ServiceFeatureTable} to the server.
+   *
+   * @param featureTable {@link ServiceFeatureTable} to apply edits to
+   */
+  private void applyEdits(ServiceFeatureTable featureTable) {
+    // apply the changes to the server
+    ListenableFuture<List<FeatureEditResult>> editResults = featureTable.applyEditsAsync();
+    editResults.addDoneListener(() -> {
+      try {
+        // check result has an edit
+        if (editResults.get().iterator().hasNext()) {
+          // attempt to get first edit from result as it should be the only edit
+          FeatureEditResult edit = editResults.get().iterator().next();
+          // check if the server edit was successful
+          if (!edit.hasCompletedWithErrors()) {
+            logToUser(getString(R.string.success_feature_deleted));
+          } else {
+            throw edit.getError();
+          }
+        }
+      } catch (InterruptedException | ExecutionException e) {
+        logToUser(getString(R.string.error_applying_edits, e.getCause().getMessage()));
+      }
+    });
+  }
+
+  /**
+   * Shows a Toast to user and logs to logcat.
+   *
+   * @param message to display to user and log to LogCat
+   */
+  private void logToUser(String message) {
+    Toast.makeText(this, message, Toast.LENGTH_LONG).show();
+    Log.d(TAG, message);
```

Comment:
should this be a log.e if its an error? And if so, should we have one function for logging if we won't always log to user in an error context?

Reply:
I've tried to reduce duplication of code. I'll refactor to allow logging to the Error level.